lib/git-runner.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+var fs = require('fs');
+
+module.exports = GitRunner;
+
+function GitRunner(config, builderOpts, commandRunner, logger) {
+  this.git = config.git;
+  this.githubOrg = builderOpts.githubOrg;
+  this.repoDir = builderOpts.repoDir;
+  this.sitePath = builderOpts.sitePath;
+  this.commandRunner = commandRunner;
+  this.logger = logger;
+}
+
+GitRunner.prototype.prepareRepo = function(branch) {
+  var gitRunner = this;
+
+  return new Promise(function(resolve) {
+    fs.exists(gitRunner.sitePath, function(exists) {
+      resolve(exists ? gitRunner.syncRepo() : gitRunner.cloneRepo(branch));
+    });
+  });
+};
+
+GitRunner.prototype.syncRepo = function() {
+  var commandRunner = this.commandRunner,
+      git = this.git;
+
+  this.logger.log('syncing repo:', this.commandRunner.repoName);
+  return commandRunner.run(git, ['stash'])
+    .then(function() {
+      return commandRunner.run(git, ['pull']);
+    })
+    .then(function() {
```

Comment:
So I was just using a 3rd party nodejs lib that was having all it's errors swallowed up because they had multiple `then` functions with no `catch`. So if an error happened in one of the `then`s or one of the promise functions, it would't propagate to node with the error. Should this also have a `catch`?